nikola/utils.py

```diff
@@ -1305,34 +1305,51 @@ class Commands(object):
     >>> commands.check(list=True)                # doctest: +SKIP
     """
 
-    def __init__(self, main):
+    def __init__(self, main, config, doitargs):
         """Takes a main instance, works as wrapper for commands."""
         self._cmdnames = []
-        for k, v in main.get_commands().items():
-            self._cmdnames.append(k)
+        self._main = main
+        self._config = config
+        self._doitargs = doitargs
+        for k, v in self._main.get_cmds().items():
```

Comment:
You should use the `cmds` provided on `doitargs`, so please don't use `DoitMain.get_cmds()` .